webpack/assets/javascripts/react_app/components/Editor/components/EditorNavbar.js

```diff
@@ -49,117 +52,138 @@ const EditorNavbar = ({
   isSelectOpen,
   showError,
   fetchAndPreview,
-}) => (
-  <div className="navbar navbar-form navbar-full-width navbar-editor">
-    <Nav className="nav nav-tabs nav-tabs-pf nav-tabs-pf-secondary">
-      <EditorRadioButton
-        stateView={selectedView}
-        btnView="input"
-        title={__('Editor')}
-        onClick={() => {
-          if (selectedView !== 'input') {
-            if (isRendering) toggleRenderView();
-            changeTab('input');
-          }
-        }}
-      />
-      <EditorRadioButton
-        stateView={selectedView}
-        disabled={!isDiff}
-        btnView="diff"
-        title={__('Changes')}
-        onClick={() => {
-          if (selectedView !== 'diff') {
-            changeTab('diff');
-          }
-        }}
-      />
-      {showPreview && (
-        <React.Fragment>
-          <EditorRadioButton
-            stateView={selectedView}
-            btnView="preview"
-            title={__('Preview')}
-            onClick={() => {
-              if (selectedView !== 'preview') {
-                if (!isRendering) toggleRenderView();
-                changeTab('preview');
-                if (selectedHost.id === '') fetchAndPreview(renderPath);
-              }
-            }}
-          />
-          {showHostSelector && (
-            <EditorHostSelect
+}) => {
+  const [selectedRenderPath, setRenderPath] = useState(renderPath);
+
+  const setSafemode = safemode => {
+    const newRenderPath = safemode ? safemodeRenderPath : renderPath;
+    setRenderPath(newRenderPath);
+    previewTemplate({ host: selectedHost, renderPath: newRenderPath });
+  };
+
```

Comment:
The `renderPath` and `safemodeRenderPath` are constant props so we don't need another state just for picking one.
we can move the `checked` state over here from `EditorSafemodeCheckbox` , and determine the chosen prop according to it, something like:

```js
 const [safemode, setSafemode] = useState(false);
 const handleSafeModeChange = ({ currentTarget: { checked: newChecked } }) => {
    setSafemode(newChecked);
    previewTemplate({ host: selectedHost, renderPath: newRenderPath });
  };
const selectedRenderPath = safemode ? safemodeRenderPath : renderPath:

return (
  // implementation
  <EditorSafemodeCheckbox
    handleChange={handleSafeModeChange}
    checked={safemode}
    {...otherProps}
 />
)
```